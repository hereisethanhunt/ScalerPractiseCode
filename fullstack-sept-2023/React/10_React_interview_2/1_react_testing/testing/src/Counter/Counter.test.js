import { fireEvent, render, screen } from '@testing-library/react';
import Counter from './Counter';


describe("counter component test case",()=>{

  test('initial values using snapshot', () => {
      const {asFragment} = render(<Counter></Counter>);
      expect(asFragment()).toMatchSnapshot();
  });
  
  test('initial state of count', () => {
    render(<Counter></Counter>);
    const increment = screen.getByText('+');
    const count = screen.getByText('Count 0');
    const decrement = screen.getByText('-');
    expect(increment).toBeInTheDocument();
    expect(count).toBeInTheDocument();
    expect(decrement).toBeInTheDocument();

});

test('update state of count by clicking on + button', () => {
  render(<Counter></Counter>);
  const increment = screen.getByText('+');
  fireEvent.click(increment);
  fireEvent.click(increment);
  fireEvent.click(increment);
  const count = screen.getByText('Count 3');
  expect(count).toBeInTheDocument();

});


})

