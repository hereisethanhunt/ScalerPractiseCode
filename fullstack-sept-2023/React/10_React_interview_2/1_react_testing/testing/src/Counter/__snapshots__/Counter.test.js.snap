// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter component test case initial values using snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button>
      +
    </button>
    <p>
      Count 0
    </p>
    <button>
      -
    </button>
    <div>
      Hello
    </div>
  </div>
</DocumentFragment>
`;
